### ISSUE TITLE
Formatting a workbook

### ISSUE THREAD

**Issue by AnnePdM** [JanMarvin/openxlsx2#383]
I have been having a problem formatting a dataset for export to Excel. I have been using the examples in the vignettes. I have been trying to format different rows of the data differently. I include a reproducible example below. In the example, the choices I have made with formatting might seem a bit odd because they are applied to a generic dataset rather than my own data. My problem is that, once I have formatted one row, this seems to affect another row.

You will note that one of my lines of code below is commented out. This line repeats the line above. The line above should embolden the header row in my table. It doesn't. Yet when repeat this command, by taking out the comment hash, the header becomes emboldened. However, in taking out the comment hash, this then affects the number formatting in the bottom line of the table. Am I doing something wrong with my code?

By way of additional information, I am using version 0.3.0.90000 of openxlsx2

```R
library(openxlsx2)

sheet <- mtcars[,c(1:6)]

wb <- wb_workbook() %>%
  wb_add_worksheet("test") %>%
  wb_add_data(x = "Title", startCol = 1, startRow = 1) %>%
  wb_add_font("test", dims = "A1", bold = "1", size = "14") %>%
  wb_add_data(x = sheet, colNames = TRUE, startCol = 1, startRow = 2, removeCellStyle = TRUE) %>%
  wb_add_cell_style("test", dims = "A2:F2", horizontal = "right") %>%
  wb_add_cell_style("test", dims = paste0("C3:D",nrow(sheet)+1), numFmtId = "2") %>%
  wb_add_cell_style("test", dims = paste0("F",nrow(sheet)+2), numFmtId = "2") %>%
  wb_add_font("test", dims = "A2:F2", bold = "1", size = "11") %>%
  #wb_add_font("test", dims = "A2:F2", bold = "1", size = "11") %>% # Not sure why I need this additional line
  wb_add_font("test", dims = paste0("A",nrow(sheet)+2,":F",nrow(sheet)+2), bold = "1", size = "11") %>%
  
  wb_add_border(dims = "A2:F2",
                bottom_color = c(theme = "1"), bottom_border = "thin",
                top_color = c(theme = "1"), top_border = "double",
                left_border = "", right_border = "") %>%
  
  wb_add_border(dims = paste0("A",nrow(sheet)+2,":F",nrow(sheet)+2),
                top_color = c(theme = "1"), top_border = "thin",
                bottom_color = c(theme = "1"), bottom_border = "double",
                left_border = "", right_border = "")

wb$open()
```

**Comment by JanMarvin** [comment]
Hi @AnnePdM ,

thanks for the report and your interest in `openxlsx2`! This might have been fixed in #365. I see the following with our latest `main`:
<img width="340" alt="Screenshot 2022-10-19 at 21 36 19" src="https://user-images.githubusercontent.com/1645626/196788506-54f47446-0a4b-4afe-9eda-ae822e4acc49.png">

Looking at your code, this is what you expected?

**Comment by AnnePdM** [comment]
Thanks for your quick reply. This is what I was expecting to see. Did you use my code to create it? How do I access the version of openxlsx2 that you used to create this?

**Comment by AnnePdM** [comment]
Thanks this seems to work but there seems to be one further idiosyncrasy which I failed to highlight in the original post. If I use the following code (line 6 of the pipe above):

`wb_add_cell_style("test", dims = "A2:F2", horizontal = "right") %>%
 `

Then all the titles right justify. However, if I wish to right justify all but the first cell it doesn't work, they all left justify:

`wb_add_cell_style("test", dims = "B2:F2", horizontal = "right") %>%
 `

Any suggestions?

**Comment by JanMarvin** [comment]
Hi @AnnePdM , thanks for asking. At first glance it seems to be a bug in `wb_add_font()`. Have to look into it later. You can move the add_font pipe above the horizontal alignment and it should work.

**Comment by JanMarvin** [comment]
Confirmed, add_font should be reworked to look like add_cell_style. Likely add_numfmt too. The issue stems from the random string named styles.

**Comment by JanMarvin** [comment]
Should be fixed in main branch once #385 is merged
